flake.nix

```diff
@@ -0,0 +1,23 @@
+{
+  description = "Home Manager for Nix";
+
+  outputs = { self, nixpkgs }: rec {
+
+    nixosModules.home-manager = import ./nixos;
+
+    lib = {
+      homeManagerConfiguration = { configuration, system, homeDirectory
+        , username
+        , pkgs ? builtins.getAttr system nixpkgs.outputs.legacyPackages
+        , check ? true }@args:
+        import ./modules nixpkgs {
```

Comment:
Does this work? The `./modules/default.nix` function takes a single argument.

Reply:
`nixpkgs` should be removed here.

Reply:
Indeed, I didn't test this part because I don't use the imperative model